sdonboarding.client/src/components/UITemplate/AddDataTemplate.jsx

```diff
@@ -0,0 +1,304 @@
+                                                        data-type='Store'
+                                                    >
+                                                        {store.name}
+                                                    </option>))}
+
+                                            </select>
+                                                ) : formElementType[index] === 'date' ? (
+                                        // Render date input if the field is a date
+                                        <input
+                                            type="date"
+                                            name={key}
+                                            id={key}
+                                            value={localObject[key]}
+                                            onChange={handleInputChange}
+                                            className="w-1/2 px-4 py-2 border rounded text-gray-400"
+                                            required
+
+                                        />
+                                    ) : formElementType[index] === 'number' ? (
+                                        // Render number input if the field is a number(Price)
+                                        <input
+                                            type="number"
+                                            name={key}
+                                            id={key}
+                                            value={localObject[key]}
+                                            onChange={handleInputChange}
+                                            className="w-full px-4 py-2 border rounded text-gray-400"
+                                            step="any" 
+                                            min="0"
+                                            placeholder={`Enter ${labels[key]}`}
+                                            required
+
+                                        />
+                                    ) : (
+                                        // Render text input for other fields (like 'name', Address)
+                                        <input
+                                            type="text"
+                                            name={key}
+                                            id={key}
+                                            value={localObject[key]}
+                                            onChange={handleInputChange}
+                                            className="w-full px-4 py-2 border rounded text-gray-400"
+                                            placeholder={`Enter ${labels[key]}`}
+                                            required
+                                        />
+                                    )}
+                                </div>
+                            );
+                        })}
+                    </div>
```

Comment:
There are too many rendering conditions in here. This will be hard and confusing to read and analyse. Instead of doing it like this, you can set up function/s and call them properly. This way, it will be easier for you and for other devs to maintain and manage the code.